ssl/s3_lib.c

```diff
@@ -4294,8 +4294,8 @@ const SSL_CIPHER *ssl3_choose_cipher(SSL_CONNECTION *s, STACK_OF(SSL_CIPHER) *cl
         maxversion = SSL_CONNECTION_IS_DTLS(s) ? c->max_dtls : c->max_tls;
 
         /* Skip ciphers not supported by the protocol version */
-        if (ssl_version_cmp(s, s->version, minversion) < 0
-            || ssl_version_cmp(s, s->version, maxversion) > 0)
+        if ((minversion > 0 && ssl_version_cmp(s, s->version, minversion) < 0)
```

Comment:
Hmmm. I'm not sure this is correct. I see a number of cases where min/max version is set to 0 for some ciphers. Notably the RC4 ciphers. These are not allowed in DTLS since they are stream ciphers - so it is correct that they should be skipped. Wouldn't this check have skipped them previously due to `ssl_version_cmp(s, s->version, minversion) < 0` being true? Now they won't be skipped.

Reply:
You are right. My mistake! I'll fix this.